server/src/controllers/jobApplicationRouter.js

```diff
@@ -18,32 +19,52 @@ jobApplicationRouter.get('/', jwtMiddleware, async (request, response) => {
       {
         model: ApplicationComment,
         as: 'applicationComments'
+      },
+      {
+        model: Attachment,
+        as: 'attachments'
       }]
   })
   response.json(jobApplications)
 })
 
 jobApplicationRouter.post('/', jobApplicationValidator, async (req, res) => {
-  const body = req.body
+  try {
+    const body = req.body
+    let attachments = []
 
-  const firstPostingStage = await PostingStage.findOne({
-    where: {
-      jobPostingId: body.jobPostingId,
-      orderNumber: 0
+    const firstPostingStage = await PostingStage.findOne({
+      where: {
+        jobPostingId: body.jobPostingId,
+        orderNumber: 0
+      }
+    })
+
+    //put to the validator?
+    if (!firstPostingStage) {
+      return res.status(400).json({ error: 'Could not find posting stage' })
     }
-  })
 
-  if (!firstPostingStage) {
-    return res.status(400).json({ error: 'Could not find posting stage' })
-  }
+    const jobApplication = await JobApplication.create({
+      applicantName: body.applicantName,
+      applicantEmail: body.applicantEmail,
+      postingStageId: firstPostingStage.id
+    })
 
-  const jobApplication = await JobApplication.create({
-    applicantName: body.applicantName,
-    applicantEmail: body.applicantEmail,
-    postingStageId: firstPostingStage.id
-  })
+    if (body.attachments.length > 0) {
+      attachments = await handleAttachmentSending(body.attachments)
+      await attachments.forEach(attachment => {
```

Comment:
`await` here probably doesn't work, as `forEach` does not return a promise, but each iteration does. However, waiting for the promises to be resolved is not needed here, as the result is not returned with the response anyway